gds_metrics/__init__.py

```
@@ -75,13 +75,19 @@ def before_request(self, *args, **kwargs):
         g._gds_metrics_start_time = monotonic()
 
     def teardown_request(self, sender, response, *args, **kwargs):
-        resp_time = monotonic() - g._gds_metrics_start_time
-        HTTP_SERVER_REQUEST_DURATION_SECONDS.labels(
-            request.method,
-            request.host,
-            request.url_rule.rule if request.url_rule else 'No endpoint',
-            response.status_code
-        ).observe(resp_time)
+        try:
+            resp_time = monotonic() - g._gds_metrics_start_time
+            HTTP_SERVER_REQUEST_DURATION_SECONDS.labels(
+                request.method,
+                request.host,
+                request.url_rule.rule if request.url_rule else 'No endpoint',
+                response.status_code
+            ).observe(resp_time)
+        except AttributeError:
+            # There is not a guarantee that `g._gds_metrics_start_time` will exist
+            # as our `before_request` function will not run if the flask app has
+            # another `before_request` that runs before ours and throws an exception
+            pass
```

Should we be doing something here in this case? Maybe a warning level log line?

---

given that this is an "expected" exception i don't think we care. it's not the loggers job to log these errors, especially if someone might make heavy use of before/after request handlers and this could be more common. i think it's fine to just pass.

---

I'm not sure I agree, I feel like that silently passing could lead to problems being harder to find later on down the line? I might be overthinking this 🤔

---

Maybe we should issue a one-time log message saying "warning: you are throwing data away" 😁